sklearn/ensemble/tests/test_iforest.py

```diff
@@ -262,14 +262,17 @@ def test_iforest_subsampled_features():
 def test_iforest_average_path_length():
     # It tests non-regression for #8549 which used the wrong formula
     # for average path length, strictly for the integer case
+    # Updated to check average path length when input is <= 2 (issue #11839)
 
     result_one = 2. * (np.log(4.) + np.euler_gamma) - 2. * 4. / 5.
     result_two = 2. * (np.log(998.) + np.euler_gamma) - 2. * 998. / 999.
-    assert_almost_equal(_average_path_length(1), 1., decimal=10)
+    assert _average_path_length(0) == 0.
+    assert _average_path_length(1) == 0.
+    assert _average_path_length(2) == 1.
```

Comment:
can we also test that `_average_path_length` is increasing for more values? I guess `_average_path_length(2) < _average_path_length(3)` would be enough